### PR TITLE
completed api/topics endpoint

### PR DESCRIPTION
**Pull request by Antony-Lester**
added files app.js, controller.js, model.js.
updated READ ME to include modules initiation & initial creation of  database

in testing covered errors for 404 bad path.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+								slug: expect.any(String),
+								description: expect.any(String),
+							}),
+						);
+					});
+				});
+		});
+		test('status:404 Not Found', () => {
+			return request(app).get('/api/topic').expect(404);
+		});
```
Comment (HeatherEver) [inline __tests__/app.test.js:40]: i would also test that you are getting back an error message you expect at this stage

File: model.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const db = require(`${__dirname}/db/connection`);
+
+function fetchTopics () {
+    return db.query(`SELECT * FROM topics;`)
+        .then(({ rows: topics }) => {return topics})
+        .catch((err) => {Promise.reject(err)})
```
Comment (HeatherEver) [inline model.js:6]: the model doesn't necessarily need a catch block as the controller will handle any errors

**Review by HeatherEver — COMMENTED** [review]
Nice work so far just a couple of things to look, mainly with the error message testing and once that is done please resubmit your pull request :)

**Comment by Antony-Lester** [comment]
catch block removed from model.js, and testing for a error message

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+						);
+					});
+				});
+		});
+		test('status:404 Not Found', () => {
+            return request(app).get('/api/topic').expect(404)
+                .then(res => {
+                    expect(res.error.path).toBe("/api/topic")
+                    expect(res.error.method).toBe("GET")
+                    expect(res.error.text).toBe("<!DOCTYPE html>\n<html lang=\"en\">\n<head>\n<meta charset=\"utf-8\">\n<title>Error</title>\n</head>\n<body>\n<pre>Cannot GET /api/topic</pre>\n</body>\n</html>\n")
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js:43]: At the minute I believe this test is looking at the default error thrown back by express, check out app.all as a means of handling a "bad path" error manually, and sending back an object with a "msg" key of your choosing

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
check out the error handling for bad path, I've left a comment on the test code

**Comment by Antony-Lester** [comment]
changed testing for error message and added app.all() catch to app.js

**Review by alexdevdavis — APPROVED** [review]
Happy to approve this, but see package.json comment as a priority

**Pull request merged**